src/App.js

```diff
@@ -1,5 +1,172 @@
+import { MissionUtils } from '@woowacourse/mission-utils';
+
+class CarRacing {
+  #cars;
+  #carsMap;
+  #numberOfTry;
+  #racingResult = '';
+
+  constructor(cars, numberOfTry) {
+    this.#cars = cars;
+    this.#numberOfTry = numberOfTry;
+    this.#carsMap = new Map(cars.map((name) => [name, '']));
+  }
+
+  race() {
+    this.#executeRacing();
+    this.#printResult();
+    this.#announceWinner();
+  }
+
+  #executeRacing() {
+    for (let i = 0; i < this.#numberOfTry; i++) {
+      this.#processOneRound();
+    }
+  }
+
+  #processOneRound() {
+    this.#updateRacingStatus();
+    this.#updateRacingResult();
+  }
+
+  #updateRacingStatus() {
+    this.#cars.forEach((name) => {
+      if (this.#shouldMoveForward()) {
+        this.#carsMap.set(
+          name,
+          this.#carsMap.get(name) + '-'
+        );
+      }
+    });
+  }
+
+  #shouldMoveForward() {
+    return MissionUtils.Random.pickNumberInRange(0, 9) >= 4;
+  }
+
+  #updateRacingResult() {
+    this.#carsMap.forEach((value, key) => {
+      this.#racingResult += `${key} : ${value}\n`;
+    });
+    this.#racingResult += '\n';
+  }
+
+  #printResult() {
+    const Console = MissionUtils.Console;
+    Console.print(this.#numberOfTry);
+    Console.print('실행 결과');
+    Console.print(this.#racingResult);
+  }
+
+  #announceWinner() {
+    const fastestDistance = this.#getFastestDistance();
+    const winners = this.#getWinners(fastestDistance);
+    MissionUtils.Console.print(
+      `최종 우승자 : ${winners.toString()}`
+    );
+  }
+
+  #getFastestDistance() {
+    return Math.max(
+      ...[...this.#carsMap.values()].map((v) => v.length)
+    );
+  }
+
+  #getWinners(fastestDistance) {
+    const winners = [];
+    this.#carsMap.forEach((v, k) => {
+      if (v === '-'.repeat(fastestDistance)) {
+        winners.push(k);
+      }
+    });
+    return winners;
+  }
+}
+
+class InputValidator {
+  static validateCarNames(inputNames) {
+    const cars = this.#parseCarNames(inputNames);
+    this.#validateInput(inputNames, cars);
+    return cars;
+  }
+
+  static #parseCarNames(inputNames) {
+    return inputNames
+      .split(',')
+      .filter((name) => name !== '')
```

Comment:
```suggestion
      .filter(Boolean)
```
빈 문자열은 falsy한 값이기 때문에. 위의 코드처럼도 사용 가능합니다! 🤗